cirq-core/cirq/transformers/gauge_compiling/gauge_compiling_test_utils.py

```diff
@@ -73,6 +74,43 @@ def test_all_gauges(self, mock_select, seed):
             else:
                 _check_equivalent_with_error_message(c, nc, gauge)
 
+    def test_sweep(self):
+        a = cirq.NamedQubit('a')
+        b = cirq.NamedQubit('b')
+        c = cirq.NamedQubit('c')
+
+        input_circuit = cirq.Circuit(
+            cirq.Moment(cirq.H(a)),
+            cirq.Moment(self.two_qubit_gate(a, b)),
+            cirq.Moment(self.two_qubit_gate(b, c)),
+            cirq.Moment(self.two_qubit_gate(a, c)),
+            cirq.Moment([cirq.H(qubit) for qubit in [a, b, c]]),
+            cirq.Moment([cirq.measure(q) for q in [a, b, c]]),
+        )
+
+        if not self.sweep_must_pass:
```

Comment:
nit: move this to the beginning of the function to avoid creating unnecessary objects

Reply:
input_circuit IS a necessary object?

Reply:
okay, I see what you mean, working on it.

Reply:
Done.